crawl-ref/source/spl-book.cc

```diff
@@ -874,6 +870,11 @@ static bool _learn_spell_checks(spell_type specspell, bool wizard = false)
 */
 bool learn_spell(spell_type specspell, bool wizard)
 {
+    if (god_loathes_spell(specspell, you.religion))
+        mprf(MSGCH_WARN,"Casting this spell will cause instant excommunication!");
+    else if (god_hates_spell(specspell, you.religion))
```

Comment:
I don't know this part of the code very well and so wasn't entirely sure on first glance that `god_loathes_spell` and `god_hates_spell` are the right ones (plus they aren't documented).  It might be worth adding a docstring to these two functions (which is a reasonable thing to do in a commit like this), and cross-checking with `spl-cast.cc:_spellcasting_god_conduct` if you haven't done so.  One more thing to double check is if this messaging makes sense for Trog, you might want something more general for that case.

Also, I might use different wording on the warnings in case punishments vary a bit by god ("Your god despises this spell" or something) but that's a matter of taste I think.